src/main/java/medic/gateway/alert/WoMessage.java

```diff
@@ -96,4 +110,23 @@ public String getFailureReason() {
 	public String toString() {
 		return String.format("%s@%s-%s", getClass().getSimpleName(), id, status);
 	}
+
+	public boolean isMaxRetriesSoftFail() {
+		return this.retries >= MAX_RETRIES_SOFT_FAIL;
+	}
+
+	/**
+	 * The wait time is incremental according with the number of retries.
+	 * @return time in milliseconds
+	 */
+	public int calcWaitTimeRetry(int retries) {
+		return WAIT_RETRY_SOFT_FAIL * retries;
```

Comment:
I think it would be better to use some sort of exponential backoff so the first retry is really quick but the 10th retry is not, for example...

`(retries ^ 2) * 60 * 1000`, ie: 1, 4, 9, 16, 25, ... minutes

Obviously this will get really big (400 minutes instead of 60 minutes) but I think that's ok - it gives more time for people to fix fixable problems, while retrying faster to get the messages out earlier for network problems.

Reply:
Okay, I could do it in the other PR that focus on this, it's easy change. One question:

By using this calculation method, Gateway will have longer waiting times between retries sooner than the previous calculation method... Once we reach the maximum number of retries, Gateway will pickup the sms again in his usual flow and try to send it, if it fails again then the retry logic starts again from the beginning: (1, 4, 9, 16... min)

While thinking on this, we were unsure of the severity/urgency of messages that Gateway receives and were feeling nervous that the current waiting time could mean "big".
Considering the current needs, Is it okay to have longer waiting times between retries sooner in timeline? 🤔

Reply:
Yes that was my concern too - my calculation was just an example. You could easily raise it to the power of 1.5, or divide the result by 2, or add a maximum so the delay is never more than 60 minutes. Let me know what you think.

Reply:
Let's try with power of 1.5 🙂, something like:
`return (int) ( Math.pow(retries, 1.5) * WAIT_RETRY_SOFT_FAIL )`

Reply:
This has been done in https://github.com/medic/medic-gateway/pull/159